### ISSUE TITLE
The BIG Gitcoin Grants CLR Round 3 Product Feedback Thread!

### ISSUE THREAD

**Issue by owocki** [gitcoinco/web#5282]
Welcome to the BIG Gitcoin Grants CLR Round 3 Product Feedback Thread!

Grants is one of our core products, but as we've been focused on [hackathons](http://hackathons.gitcoin.co) it's the one we haven't had much time to spend devving on as much recently. 

The next big effort to improove grants will be CLR round 4 (probably in Q4 sometime).

This is a thread where you can catalogue product feedback for us to work on next time.  

TLDR:
- Love Round 3? Post below
- Found a bug? Post below
- Got a comment? Post below

**Comment by owocki** [comment]
grants idx page is slow; needs some cacching/optimizationn b4 round 4

**Comment by owocki** [comment]
i think dominant assurance contracts would be cool https://github.com/gitcoinco/web/issues/4158

**Comment by nanexcool** [comment]
I loved it! My main gripe is: I want to donate to 5 projects, need to do 10 txs. How about approving once and giving out from that allowance?

**Comment by kobuta23** [comment]
Some projects have complained about difficulty listing onto the platform. Using an on-chain listing process (ex. Kleros TCR) instead of a manual whitelisting would be cool

Some clear rules about what is defined as collusion/wrongdoing would be nice! I understand identifying collusion is hard enough as it is. Defining it clearly would be a good first step IMO

Also, not everyone has a GitHub account. Some users have approached me asking if there was a different way to authenticate their humanness. I understand GitCoin uses GitHub, but having an alternative (at least for the grants) would be hugely beneficial in bringing in new users!

**Comment by nanexcool** [comment]
One other thing! It would be great if you could verify this contract on Etherscan

https://etherscan.io/address/0xdf869fad6db91f437b59f1edefab319493d4c4ce#code

I decompiled and it's pretty simple, but would still be good for adding trust. This is the one to donate an ERC20 one, not the recurring contracts. Although from what I saw, in the case of recurring payments, each project has its own contract and they're also not verified.

**Comment by owocki** [comment]
need to reduce approve() collisoins for contributions that happen right after each other 

ex https://etherscan.io/tx/0x055503218c735e7b88f5c013c21cadde45e7314071de68c066dcbb6c3afce7a2

**Comment by LefterisJP** [comment]
I may have stumbled across a bug so perhaps a different issue should be in order.

1. I don't like giving the option to any extension like metamask to have access to all sites so I had it configured to need permission given by me. I had given it permission before browsing the grants page.

2. I tried to fund my first grant, but clicking the "Fund" button was doing nothing. After 2-3 clicks I realized it may be a problem with metamask permissions so I enabled it to have  access on all sites.

3. Once I did that the metamask popup windows opened immediately. I assume they were from the click I made before and I just confirmed. I remember reading more than one confirmation is needed so I kind of blindly clicked all 4 confirmations.

4. I am not sure exactly what happened but the result is that one of the transactions was rejected: https://etherscan.io/tx/0x3667ce90ed31fd6b66856ebabd81a0dcad11fbe45fa1cc543d2b0a59acae4943 but still the grants UI shows that I contributed 1 DAI to defiproject, 2 times while from the token transfers it seems the second DAI did not go through.
![2019-10-01-122906_1119x686_scrot](https://user-images.githubusercontent.com/1658405/65956542-e60f8380-e44a-11e9-83ba-e956f9f825ff.png)

------

Another interesting problem is that even though I always chose 1 DAI of contribution, 2/5 transactions had 0.95 DAI contribution and 0.05 gitcoin cut, while 3/5 transactions had 1 DAI contribution and 0.05 gitcoin cut.

![2019-10-01-125723_1201x516_scrot](https://user-images.githubusercontent.com/1658405/65956655-2c64e280-e44b-11e9-8acd-4913a89b658f.png)

**Comment by owocki** [comment]
people keep contacting my on twitter asking why their typeform grant submissions were not accepted and why they never heard back. we need a better way of managing the typeform queue next time .

cc @ceresstation
